snuba/migrations/backfill_errors.py

```diff
@@ -6,462 +6,196 @@
-            "ip_address_v6",
-            "user",
-            "user_id",
-            "user_name",
-            "user_email",
-            "sdk_name",
-            "sdk_version",
-            "http_method",
-            "http_referer",
-            "tags.key",
-            "tags.value",
-            "contexts.key",
-            "contexts.value",
-            "transaction_name",
-            "span_id",
-            "trace_id",
-            "partition",
-            "offset",
-            "message_timestamp",
-            "retention_days",
-            "deleted",
-            "group_id",
-            "primary_hash",
-            "received",
-            "message",
-            "title",
-            "culprit",
-            "level",
-            "location",
-            "version",
-            "type",
-            "exception_stacks.type",
-            "exception_stacks.value",
-            "exception_stacks.mechanism_type",
-            "exception_stacks.mechanism_handled",
-            "exception_frames.abs_path",
-            "exception_frames.colno",
-            "exception_frames.filename",
-            "exception_frames.function",
-            "exception_frames.lineno",
-            "exception_frames.in_app",
-            "exception_frames.package",
-            "exception_frames.module",
-            "exception_frames.stack_level",
-            "sdk_integrations",
-            "modules.name",
-            "modules.version",
-        ]
+def get_monday(day: date) -> date:
+    return day - timedelta(days=day.weekday())
 
 
-def process_event(event_data: Sequence[Any]) -> Mapping[str, Any]:
-    event = Event(*event_data)
-    ip_v4 = None
-    ip_v6 = None
-    ip_address = _ensure_valid_ip(event.ip_address)
+WINDOW = timedelta(days=7)
+BEGINNING_OF_TIME = get_monday((datetime.utcnow() - timedelta(days=90)).date())
 
-    if ip_address:
-        if ip_address.version == 4:
-            ip_v4 = str(ip_address)
-        elif ip_address.version == 6:
-            ip_v6 = str(ip_address)
 
-    # Extract promoted contexts
-    trace_id = None
-    span_id = None
-
-    try:
-        trace_idx = event.contexts_key.index("trace")
-        transaction_ctx = json.loads(event.contexts_value[trace_idx])
-    except ValueError:
-        transaction_ctx = {}
-
-    if transaction_ctx.get("trace_id", None):
-        trace_id = UUID(transaction_ctx["trace_id"])
-    if transaction_ctx.get("span_id", None):
-        span_id = int(transaction_ctx["span_id"], 16)
+contexts_key = """
+arrayMap(
+    t -> tupleElement(t, 1),
+    arraySort(
+        arrayFilter(
+            t -> tupleElement(t, 2) IS NOT NULL,
+            arrayConcat(
+                arrayZip(contexts.key, contexts.value),
+                [('geo_city', geo_city)],
+                [('geo_country_code', geo_country_code)],
+                [('geo_region', geo_region)],
+                [('os_build', os_build)],
+                [('os_kernel_version', os_kernel_version)]
+            )
```

Comment:
Still missing a bunch of fields from the device context.
https://github.com/getsentry/snuba/blob/master/snuba/datasets/events_processor.py#L117-L129

Reply:
Good point, thanks